### PR TITLE
Fix/condo/doma 2354/fix ticket creation by mobile app

### PR DESCRIPTION
**Pull request by Kanol**
Problem:
If ticket created from mobile app, it doesn't have data in contact field. This is why due ti ticket update, it overwrite contact field
Solution:
Add contact field to ticket creation

### CODE REVIEW

**Review by pahaz — APPROVED** [review]

**Review by dkoviazin — APPROVED** [review]

**Review by AleX83Xpert — APPROVED** [review]

**Review by akarjakin — APPROVED** [review]

**Review by sitozzz — APPROVED** [review]

File: apps/condo/domains/ticket/schema/CreateResidentTicketService.test.js [hunk 1]
```diff
@@ -90,6 +92,23 @@ describe('CreateResidentTicketService', () => {
...
         expect(contact.id).toMatch(UUID_RE)
     })
+    test('resident: created ticket has contact', async () => {
+        const admin = await makeLoggedInAdminClient()
+        const userClient = await makeClientWithResidentAccessAndProperty()
+        const [residentTicket] = await createResidentTicketByTestClient(userClient, userClient.property)
+        const [ticket] = await Ticket.getAll(admin, {
+            id: residentTicket.id,
+        })
+        const [contact] = await Contact.getAll(admin, {
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/CreateResidentTicketService.test.js:102]: why not get one?

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 2]
```diff
@@ -333,6 +334,26 @@ const Ticket = new GQLListSchema('Ticket', {
                 resolvedData.propertyAddress = property.address
                 resolvedData.propertyAddressMeta = property.addressMeta
             }
+            if (!resolvedData.contact) {
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.js:337]: ??? we don't need to create a contact automatically ???

**Review by pahaz — CHANGES_REQUESTED** [review]

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 3]
```diff
@@ -346,16 +367,16 @@ const Ticket = new GQLListSchema('Ticket', {
                 if (resolvedData.statusUpdatedAt) {
                     if (existingItem.statusUpdatedAt) {
                         if (new Date(resolvedData.statusUpdatedAt) <= new Date(existingItem.statusUpdatedAt)) {
-                            return addValidationError(`${ STATUS_UPDATED_AT_ERROR }statusUpdatedAt] Incorrect \`statusUpdatedAt\``)
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.js:349]: redundant changes

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 4]
```diff
@@ -346,16 +367,16 @@ const Ticket = new GQLListSchema('Ticket', {
...
+                            return addValidationError(`${STATUS_UPDATED_AT_ERROR}statusUpdatedAt] Incorrect \`statusUpdatedAt\``)
                         }
                     } else {
                         if (new Date(resolvedData.statusUpdatedAt) <= new Date(existingItem.createdAt)) {
-                            return addValidationError(`${ STATUS_UPDATED_AT_ERROR }statusUpdatedAt] Incorrect \`statusUpdatedAt\``)
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.js:353]: rollback it please

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 5]
```diff
@@ -346,16 +367,16 @@ const Ticket = new GQLListSchema('Ticket', {
...
+                            return addValidationError(`${STATUS_UPDATED_AT_ERROR}statusUpdatedAt] Incorrect \`statusUpdatedAt\``)
                         }
                     }
                 }
             } else {
-                return addValidationError(`${ DV_UNKNOWN_VERSION_ERROR }dv] Unknown \`dv\``)
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.js:358]: why do you need to change this line?

File: apps/condo/domains/ticket/schema/CreateResidentTicketService.test.js (hunk 1 above)
Comment (Kanol) [reply apps/condo/domains/ticket/schema/CreateResidentTicketService.test.js:102]: Getall with where condition

File: apps/condo/domains/ticket/schema/Ticket.js (hunk 2 above)
Comment (Kanol) [reply apps/condo/domains/ticket/schema/Ticket.js:337]: If contact data was sended, I don't think we should overwrite it

**Pull request closed without merging**